disease_gene/disease_associates_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,491 @@
+
+# In[20]:
+
+
+sns.distplot(total_candidates_df.query("sen_length < 83+1")["sen_length"], rug=False)
+
+
+# In[21]:
+
+
+total_candidates_df.query("sen_length < 83+1")["sen_length"].describe().astype(int)
+
+
+# This distribution looks a bit more reasonable compared to the above distribution. After filtering out the outliers, we still have a pleathora of sentences on the order of 3.6 million. (removed 146841 sentences).
+
+# In[22]:
+
+
+before_filter = set([tuple(line) for line in total_candidates_df[["entrez_gene_id", "doid_id"]].values])
+after_filter = set([tuple(line) for line in total_candidates_df.query("sen_length < 83+1")[["entrez_gene_id", "doid_id"]].values])
+print(
+    "Total number of entity candidates before filter: {}".format(
+        total_candidates_df[["entrez_gene_id", "doid_id"]].drop_duplicates().shape[0]
+    )
+)
+print("Total number of entity candidates being thrown out: {}".format(len(before_filter.difference(after_filter))))
+
+
+# In[23]:
+
+
+filtered_total_candidates_df = total_candidates_df.query("sen_length < 83+1")
+
+
+# In[24]:
+
+
+venn2(
+    [
+        set(filtered_total_candidates_df.query("hetionet==0").sentence_id),
+        set(filtered_total_candidates_df.query("hetionet==1").sentence_id)
+    ], set_labels=["Not In Hetionet", "In Hetionet"])
+plt.title("# of Unique Sentences in Entire Dataset with Co-Mention Pair in/not in hetionet")
+
+
+# # Co-occuring Mentions Sentence Stats
+
+# This next block contains a visualization about the number of mentions a sentence may contain.
+
+# In[25]:
+
+
+ax = sns.scatterplot(
+    x="Disease_mention_count", 
+    y="Gene_mention_count", 
+    data=
+    (
+        filtered_total_candidates_df
+        .drop_duplicates("sentence_id")
+        .groupby(["Disease_mention_count", "Gene_mention_count"]).size()
+        .reset_index()
+        .rename(index=str, columns={0:"size"})
+        .assign(natural_log_size= lambda x: pd.np.log(x['size']))
+    ),
+    hue="natural_log_size",
+    palette='viridis'
+)
+ax.set_title("Mention Distribution of Entire Dataset")
+
+
+# In the above graph every point is group of possible mention counts sentences could contain. For example one group has sentences containing one gene mention and one disease mention (indicated in the bottom left in yellow). Overall this graph is acceptable in terms of the spread between mentions; however, when getting ready to train the deep learning models one has to make sure the algorithm can detect which co-mention pair the candidate is referring to.
+
+# # Subsampled Set Distributions
+
+# Take a look at the stratification for each train/dev/test set.
+
+# ## Training Set
+
+# Here the training set covers majority of the characterisitics are the entire dataset. This isn't surprising because a stratified sort was used to allocate candidates.
+
+# In[26]:
+
+
+ax = sns.scatterplot(
```

Comment:
To clarify, for one of the dark blue plots, is that the number of sentences that have 25 disease mentions and 0 gene mentions?  So its 3D?

Reply:
In this plot every circle is a group of sentences based on the number of mentions contained in that sentence. (e.x. The top leftmost circle is a group where sentences have ~1 disease mention and ~45 different gene mentions.) The color shows how many sentences belong to each given group. The brighter the color the more sentences appear in that group.